src/mito2/src/worker.rs

```diff
@@ -563,6 +567,30 @@ impl<S: LogStore> RegionWorkerLoop<S> {
             BackgroundNotify::CompactionFailed(req) => self.handle_compaction_failure(req).await,
         }
     }
+
+    /// Handles internal command.
+    async fn handle_inter_command(&mut self, region_id: RegionId, command: Command) {
+        match command {
+            Command::SetReadonlyGracefully(req) => {
+                if let Some(region) = self.regions.get_region(region_id) {
+                    if region.is_writable() {
+                        region.set_writable(false);
+                    }
```

Comment:
Same result as setting false directly.
```suggestion
                    region.set_writable(false);
```